index.md

```diff
@@ -213,34 +213,34 @@ and give you credit for your work.
+2.  `data` contains raw data and metadata, organized into subdirectories if needed. Note that this directory contains only unprocessed data, cleaned or otherwise modified data files are considered a result.
+3.  `src` contains the source code for scripts and programs, which may be written in interpreted languages such as R or Python or compiled languages like Fortran, C++, or Java.
+4.  `bin` contains executable scripts and programs that are brought in from other sources or compiled from code in the `src` directory. Projects that use only modern interpreted languages, such as R or Python, will not require this directory.
+5.  `results` for all files that are generated as part of the project. This includes both intermediate results, such as cleaned data sets or simulated data, as well as final results such as figures and tables.
 
 ### Discussion
 
-FIXME
+The `src` directory often contains two conceptually distinct types of files that should be distinguished either by clear file names or by additional subdirectories. The first type are individual files, or related groups of files, that contain functions to perform the core analysis of the research. If there are more than one or two of these files, these should be organized into additional subdirectories following language-specific rules or conventions (e.g., C++ header files, Python or R package structure).
+
+A second type of file in this directory are controller or driver scripts that combine the core analytical functions with specific analysis parameters and data input and output commands in order to execute the entire project analysis, from start to finish. A simple controller script, for example, may read in a raw data table, import and apply several analysis functions from the other files in this directory, and create and save a figure. For a small project with one main output, a single controller script should be placed in the main `src` directory and distinguished clearly from the other scientific code by a name such as "runall".
+
+The `results` directory will also generally require additional structure for all but the simplest projects. At a minimum, intermediate files such as cleaned data, statistical tables, and final publication-ready figures or tables should be separated clearly by file naming conventions or placed into different subdirectories.
+
+In an ideal project, a controller script should be able to create all of the results found in the `results` directory automatically, using the contents of `data`, `src`, and/or `bin`, with no manual human intervention. This helps to ensure both provenance, so that results can always be associated with the upstream files that generated them, and reproducibility, so that you or others can recreate the outputs of any analysis. If this goal is achieved, then the contents of the `results` directory do not need to be placed under version control (see below), as they do not contain any unique information.
```

Comment:
Which scripts are you imagining? The `runall` script is just that - are you thinking of naming conventions for other scripts that it would invoke for various stages of the analysis? Honestly in projects at this level, probably anything logical will do since there's unlikely to be more than 10 or so (so you could use `fig_count_histogram` and `fig_scatter_raw_data` or whatever) - I can suggest something to that effect.